app/policies/participant_profile/ecf_policy.rb

```diff
@@ -47,8 +47,20 @@ def withdraw_record?
 
   def same_school?
     InductionRecord.joins(:school)
-      .where(school: { id: user.induction_coordinator_profile.schools.select(:id) })
+      .where(school: { id: sit_school_ids })
       .where(participant_profile_id: record.id)
       .any?
   end
+
+  def mentoring_at_school?
+    record.mentor? && sit_school_ids.intersect?(mentor_school_ids)
+  end
+
+  def sit_school_ids
+    user.induction_coordinator_profile.schools.select(:id).map(&:id)
+  end
+
+  def mentor_school_ids
+    record.school_mentors.select(:school_id).map(&:school_id)
+  end
```

Comment:
How does this work if the SIT is at multiple schools and the mentor is at a different school to the one being viewed?

Reply:
@tonyheadford the SIT will only see the mentor listed in the "Manage mentors and ECTs" page, when they are viewing School B.